### PR TITLE
feat(vue): support SSR

### PR DESCRIPTION
**Pull request by christian-bromann**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally for affected output targets
- [ ] Tests (`npm test`) were run locally and passed
- [x] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Currently the Vue Output Target would not support SSR.

## What is the new behavior?

This patch adds support for it. Furthermore:

- it adds a Nuxt.js example application for testing purposes
- it requires users to have `@stencil/vue-output-target` as dependency to access runtime files (similar to React)
- it adds a `hydrateModule` property to the output target that allows users to enable SSR

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: packages/example-project/component-library/stencil.config.ts [hunk 1]
```diff
@@ -66,8 +66,12 @@ export const config: Config = {
       hydrateModule: 'component-library/hydrate'
     }),
     vueOutputTarget({
```
Comment (tanner-reits) [inline packages/example-project/component-library/stencil.config.ts:68]: Should we have two Vue output target configs like we do for React?

**Review by tanner-reits — APPROVED** [review]
There's a lot here, but looks like it works as expected. I tested out the Nuxt app and everything was rendering (some of the components are goofy, but that's just the components, not the SSR implementation). I didn't double-check a client-side Vue app.

**Comment by christian-bromann** [comment]
> I didn't double-check a client-side Vue app.

FWIW I raised PR adding an example app using Vite + Vue, see https://github.com/ionic-team/stencil-ds-output-targets/pull/499. For that I've added an e2e test for this. For all these environments we now run a WebdriverIO tests that verifies that Stencil is properly able to hydrate the component, e.g. interacting with an input component with SSR support or without. While these tests are very rudimental, they are a start a least 😉

File: packages/example-project/component-library/stencil.config.ts (hunk 1 above)
Comment (christian-bromann) [reply packages/example-project/component-library/stencil.config.ts:68]: React and Vue projects are setup differently here. We have two output targets for React because we directly compile the components into their represented projects. With Vue we have setup an example project that just re-exports the components, and potentially more Vue related modules. That example project is imported by Nuxt and the Vite project. A bit different setup but it is actually how we see projects being organised.

File: packages/example-project/nuxt-app/public/robots.txt
Comment (thetaPC) [inline packages/example-project/nuxt-app/public/robots.txt:1]: Is this file needed?

File: packages/vue-output-target/src/generate-vue-component.ts [hunk 2]
```diff
@@ -1,34 +1,57 @@
...
+    ${Object.entries(propMap)
+      .map(([key, [type, attr]]) => (attr ? `'${key}': [${type}, "${attr}"]` : `'${key}': [${type}]`))
+      .join(',\n    ')}
+  }
+})`
+      : '';
+
+    // tagName: string;
+    // hydrateModule: Promise<{ renderToString: RenderToString }>;
+    // props?: Record<string, any>;
```
Comment (thetaPC) [inline packages/vue-output-target/src/generate-vue-component.ts:51]: These should be removed if they aren't going to be used.

File: packages/example-project/nuxt-app/README.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+# pnpm
+pnpm run dev
+
+# yarn
+yarn dev
+
+# bun
+bun run dev
+```
```
Comment (thetaPC) [inline packages/example-project/nuxt-app/README.md:39]: I'm getting the following error when trying to run it locally: `Missing script: "dev"`. These commands need to be changed to use `start` instead of `dev`.

File: packages/example-project/nuxt-app/public/robots.txt
Comment (christian-bromann) [reply packages/example-project/nuxt-app/public/robots.txt:1]: No, it was part of the basic Nuxt setup. I can remove it.

File: packages/example-project/nuxt-app/README.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+# pnpm
+pnpm run preview
+
+# yarn
+yarn preview
+
+# bun
+bun run preview
+```
```
Comment (thetaPC) [inline packages/example-project/nuxt-app/README.md:73]: I'm getting this error when trying to visit the preview:
```
Listening on http://[::]:3000
[nuxt] [request error] [unhandled] [500] Cannot find module '/Users/maria/Desktop/projects/stencil-ds-output-targets/packages/example-project/nuxt-app/.output/server/node_modules/vue/server-renderer/index.mjs'
```

**Review by thetaPC — APPROVED** [review]
LGTM, added some comments but they're not blockers
